test/integration/scheduler/preemption_test.go

```diff
@@ -1023,6 +1024,93 @@ func TestNominatedNodeCleanUp(t *testing.T) {
 	}
 }
 
+func TestNominatedNodeCleanUpUponNodeDeletion(t *testing.T) {
```

Comment:
This test can be merged with existing `TestNominatedNodeCleanUp` using table-test style.